bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend

```diff
@@ -160,23 +161,31 @@ class GenericThingProvider extends AbstractProvider<Thing> implements ThingProvi
             return new ThingTypeUID(thingUID.bindingId, thingUID.thingTypeId)
         }
     }
-
-    def private void createThing(ModelThing modelThing, Bridge parentBridge, Collection<Thing> thingList,
-        ThingHandlerFactory thingHandlerFactory) {
-        val ThingUID thingUID = getThingUID(modelThing, parentBridge?.UID)
+    
+    def private Iterable<ModelThing> flattenModelThings(Iterable<ModelThing> things) {
+        if (things === null || things.length === 0) {
+            return #[]
+        }
+        
+        // construct the thingUID up front: this will ensure that short notation things with reference to a bridge 
+        // do not end up with the bridge id in their id. 
+        things.forEach([thingId = thingId ?: constructThingUID.toString])
+        things.filter(typeof(ModelBridge)).forEach([val b = it; b.things.forEach[
```

Comment:
~~`val b = it;` -> `b |`~~

Nevermind, all correct. Still, here you got me - how about giving _both_  `it`s some explicit names?